### PR TITLE
shared rollup config

### PR DESCRIPTION
**Pull request by romainmenke**
- add a shared rollup config for both javascript and typescript.
- this should support `esm`, `cjs` and cli's.

Notes :

1) Some dependencies have circular imports and rollup nags about these.
In practice they seem to work fine so I silence them after testing.

2) Typescript support is added early here as I think it is an important tool.
There is no plan to migrate everything to Typescript now. It should just be possible to use in this repo.

### CODE REVIEW

File: rollup/default.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+		plugins: [
+			babel({
+				babelHelpers: 'bundled',
+				exclude: 'node_modules/**',
+				presets: [
+					['@babel/preset-env', {
+						corejs: 3,
+						loose: true,
+						modules: false,
+						targets: { node: 12 },
```
Comment (Antonio-Laguna) [inline rollup/default.js:34]: Are targets correct here? The current LTS is 16

File: rollup/default.js [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			terser(),
+		],
+	},
+	{
+		input: 'src/cli.js',
+		output: [
+			{ file: 'dist/cli.mjs', format: 'esm', sourcemap: false },
+		],
+		onwarn: (warning) => {
+			// Silence circular dependency warning for postcss-values-parsers package
```
Comment (Antonio-Laguna) [inline rollup/default.js:48]: Can you outline where the circular dependency is? I'd love to get rid of this workaround if doable

File: rollup/default.ts.js [hunk 3]
```diff
@@ -0,0 +1,86 @@
+import babel from '@rollup/plugin-babel';
+import commonjs from '@rollup/plugin-commonjs';
+import path from 'path';
+import typescript from '@rollup/plugin-typescript';
+import { nodeResolve } from '@rollup/plugin-node-resolve';
+import { terser } from 'rollup-plugin-terser';
+
+export default [
+	{
+		input: 'src/index.ts',
```
Comment (Antonio-Laguna) [inline rollup/default.ts.js:10]: Is there a way for us to reuse all of this that's repeated here so it's common and easier to maintain?

**Review by Antonio-Laguna — COMMENTED** [review]
This is great and I love you've opened the door to TS globally :) 

Some comments, happy to discuss. Most of this would clash with `;` for example on eslint

File: rollup/default.ts.js (hunk 3 above)
Comment (romainmenke) [reply rollup/default.ts.js:10]: Definitely but it might be too early for that.
I first want to see that these configs hold up for every plugin.
Only did some spot tests so I don't yet have a good view of all edge cases.

Having everything explicit makes it easier now to handle the unexpected.

I'll make an issue to clean this up later.

File: rollup/default.js (hunk 1 above)
Comment (romainmenke) [reply rollup/default.js:34]: I think so, but now starting to doubt it.
I thought we declare node 12 here as that is the lowest version we actively support.
Otherwise the build output might not work in node 12|14

https://babeljs.io/docs/en/babel-preset-env#targetsnode

File: rollup/default.js (hunk 2 above)
Comment (romainmenke) [reply rollup/default.js:48]: I opened an issue to outline this later once the general setup is complete : https://github.com/csstools/postcss-plugins/issues/8

At the moment I can't run it to trigger the warning again.

File: rollup/default.js (hunk 1 above)
Comment (Antonio-Laguna) [reply rollup/default.js:34]: Makes sense

File: rollup/default.js (hunk 2 above)
Comment (Antonio-Laguna) [reply rollup/default.js:48]: Thanks for doing that!

File: rollup/default.ts.js (hunk 3 above)
Comment (Antonio-Laguna) [reply rollup/default.ts.js:10]: Perfect!
